ts/components/wallet/card/Logo.tsx

```diff
@@ -21,15 +21,35 @@ export const cardIcons: { [key in CreditCardType]: any } = {
   UNKNOWN: require("../../../../img/wallet/cards-icons/unknown.png")
 };
 
-// PagoPA's "brandLogo" is not a string that
-// allows for the identification of a logo,
-// but rather a link to the card logo image
-// As a temporary fix, the "unknown" card will be
-// shown for all cards -- a future story will take
-// care of switching the images to the actual logos
-// @https://www.pivotaltracker.com/story/show/159651239
-const getCardIconFromBrandLogo = (_: Wallet) => {
-  return require("../../../../img/wallet/cards-icons/unknown.png");
+const cardMapIcon: { [key in string]: any } = {
+  carta_mc: require("../../../../img/wallet/cards-icons/mastercard.png"),
+  carta_visa: require("../../../../img/wallet/cards-icons/visa.png"),
+  carta_amex: require("../../../../img/wallet/cards-icons/amex.png"),
+  carta_diners: require("../../../../img/wallet/cards-icons/diners.png"),
+  carta_visaelectron: require("../../../../img/wallet/cards-icons/visa-electron.png"),
+  carta_poste: require("../../../../img/wallet/cards-icons/postepay.png")
+};
+
+/**
+ * Return the base name of a remote resource from a give path
+ * @param resourceUrl the remote resource Url
+ */
+const getImageNameFromUrl = (resourceUrl?: string): string | undefined => {
```

Comment:
instead of accepting an `undefined` value and returning `undefined` in this function, do that check at the caller site and make this function accept a `string`

Reply:
move this method somewhere under `utils` as it's not specific to this component